src/features/devices/hooks/defaults-validation.ts

```diff
@@ -58,15 +58,6 @@ hooks.addPureHook('PATCH', 'resin', 'device', {
 			}
 		}
 
-		// When moving application make sure to set the build to null, unless a specific new
-		// build has been targeted, instead of pointing to a build of the wrong application
-		if (
-			request.values.belongs_to__application != null &&
-			request.values.should_be_running__release === undefined
-		) {
-			request.values.should_be_running__release = null;
```

Comment:
Is this not needed anymore? I don't see an equivalent hook on `device_application`.

Reply:
It won't be needed when it's on device_application itself but for now it is, fixed